### PR TITLE
Request to merge into the Main Branch.

### PR DESCRIPTION
**Pull request by AdedayoOpeyemi**
Pull request to merge into Main branch.

## Project specifications

    APIs

          First, you need to find an API so you can base the development of the webapp around it. The API should allow you to:

            Get a list of items with a unique item id (or generate the unique id).
            For a given item, get detailed information about it.

          We recommend that you choose an API that doesn't need authentication. if you select an API that requires authentication, you should implement it on your own. Also, if you select an API that provides image resources your webapp will be more visual.

          Some example APIs are:

            Pokeapi: data about the Pokémon world.
            TVmaze API: data about TV series and movies.
            Meals DB: data about meals.

          You can find more APIs in this GitHub repo or in ProgrammableWeb. Some of the APIs require a token, some require authentication, and some others are just open.

            You will use our Involvement API to record the different user interactions (likes, comments and reservations).

    Interfaces

          You should build these interfaces:
                The home page.
                The comments popup.
                The reservations popup (only for the groups of 3 students).
          You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.

          Home page
            When the page loads, the webapp retrieves data from:
              The selected API and shows the list of items on screen.
              The Involvement API to show the item likes.
            Remember that your page should make only 2 requests:
              One to the base API.
              And one to the Involvement API.
            When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.
            When the user clicks on the "Comments" button, the Comments popup appears.
            When the user clicks on the "Reservations" button, the Reservations popup appears (only for the groups of 3 students).
            Home page header and navigation similar to the given mockup.
            Home page footer similar to the given mockup.

          Comments popup
            When the popup loads, the webapp retrieves data from:
              The selected API and shows details about the selected item.
              The Involvement API to show the item comments.
            When the user clicks on the "Comment" button, the data is recorded in the Involvement API and the screen is updated.

          Reservations popup (only for the groups of 3 students)
            When the popup loads, the webapp retrieves data from:
              The selected API and shows details about the selected item.
              The Involvement API to show the item reservations.
            When the user clicks on the "Reserve" button, the data is recorded in the Involvement API and the screen is updated.

          Counters We have counters in all the interfaces that show:

            The number of items (home).
            The number of comments (comments popup).
            The number of reservations (reservations popup) - only for the groups of 3 students.

          Even if the API gives you these numbers, you will create a specific function to calculate these numbers in each page. These count functions need to be covered with unit tests using Jest.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  renderShows(allShows, allLikes)
+    .then(() => {
+      imageListener();
+      commentsListener();
+      reservationListener();
+      likesListener();
+      const output = numberOfShows(allShows);
+      addNumberOfShows(output);
+    });
+});
```
Comment (oxenprogrammer) [inline src/index.js]: **Changes Required**
- [x] Fair effort on using the `async await`  promise syntactic sugar. However, you're mixing two syntax here, and the `async` keyword in this context is redundant. May we please remove the `.then` and stick to the `async await` format.

File: src/modules/ModalInterface.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+   </div>
+    <h3 class="modal-auxiliar-titles mb-3">Comments</h3>
+   <div id="comments-container" class="d-flex flex-column mb-3">
+   </div>
+   <h3 class="modal-auxiliar-titles mb-3">Add a comment</h3>
+   <form action="submit" class="d-flex flex-column">
+     <input type="text" class="modal-input mb-2" placeholder="Your Name"/>
+     <textarea class="modal-textarea mb-2" placeholder="Your comment" cols="30" rows="10"></textarea>
+     <button type="submit" class="default-button mb-2">Comment</button>
+   </form>
```
Comment (oxenprogrammer) [inline src/modules/ModalInterface.js:36]: **Optional Change**
- [x] Getting `404` when I try to make a comment as shown below. Let us try to fix this.

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Hi @AdedayoOpeyemi , 

Good job so far! to highlight,
- [x] Professional README file
- [x] Correct gitflow
- [x]  Nice UI

I can see you're really putting effort to become a world-class software developer :+1:
There are some issues that you still need to address in order to move forward to the next project.
Trust me you got this! :muscle:


## Required Changes ♻️
_Check the comments under the review._

**Changes Required**
- [ ] The Video presentation is currently not loading. Please ensure that it loads and then attach the link to your README file.

## Optional suggestions
_Consider the optional changes nice to have features and only work on them when you have satisfied the required changes_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/index.js (hunk 1 above)
Comment (AdedayoOpeyemi) [reply src/index.js]: Thanks.

Changes made.

File: src/modules/ModalInterface.js (hunk 2 above)
Comment (AdedayoOpeyemi) [reply src/modules/ModalInterface.js:36]: Thanks for pointing this out.

However, this is more like a solo project as I was unpaired with my partner sometime during the project and this task falls under the requirements for my previous partner. 

For purpose of review, I am person A and already did all tasks for person A

File: src/index.js (hunk 1 above)
Comment (henatan99) [reply src/index.js]: This issue is well addressed. Well done! :clap:

File: src/modules/ModalInterface.js (hunk 2 above)
Comment (henatan99) [reply src/modules/ModalInterface.js:36]: I have checked the comments functionalities tasks are assigned to your partner.  As a result, you are free to omit that part. But, I highly recommend you complete the project when you get time as it showcases your skills to potential employers. Also I would love to see if you are formally unpaired with your partner. For now, I will approve your project.

**Review by henatan99 — APPROVED** [review]
Hi @AdedayoOpeyemi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**